File: Data Structures & Algorithms/missing-number/submission-0.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int res=-1;
        for(int num:nums){
            res = res^num;
        }
        int j = 0; 
        while(j<=nums.size()){
            res=res^j; 
            j++; 
        }
        return res^-1; 
    }
};

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        result = []
        combo=[]

        def helper():
            if len(combo)==len(nums):
                result.append(combo[:])
                return
            for num in nums:
                if not num in combo:
                    combo.append(num)
                    helper()
                    combo.pop()
        helper()
        return result